### DISCUSSION TITLE
How to register global Directive?

### DISCUSSION THREAD

**Discussion by yunyuyuan** [nuxt/nuxt#17573] [Questions] [answered]
I didn't find the right solution. I have tried this:
```js
// plugin/xxx.client.ts

export default defineNuxtPlugin((app: NuxtApp) => {
  app.vueApp.directive('v-viewer', {
    mounted(el) {
      console.log('mounted', el);
    },
  });
})
```

But get an error: `Cannot read properties of undefined (reading 'getSSRProps')`.Maybe it because I wan't the directive been used in SSR,but Nuxt dosen't know how to compile this directive ?

**Comment by yunyuyuan** [comment, marked as answer]
Solved.
* In vue3,directive are automatically prefixed with `v`
* Should use `<client-only>`

```js
// plugin/xxx.client.ts

export default defineNuxtPlugin((app: NuxtApp) => {
  app.vueApp.directive('viewer', {
    mounted(el) {
      console.log('mounted', el);
    },
  });
})
```
```html
// pages/xxx.vue

<template>
    ......
   <client-only>
      <div v-viewer>
          <img  src=""/>
      </div>
  </client-only>
    ......
</template>
```

**Reply by Der-Alex** [reply]
You could also register the directive server side. the filename has to be `plugin/xxx.ts` then. on template side you don't have to use `<client-only>` anymore.
